types/canonical.go

```diff
@@ -3,15 +3,65 @@ package types
 import (
 	"time"
 
+	cmtbytes "github.com/cometbft/cometbft/libs/bytes"
 	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
 	cmttime "github.com/cometbft/cometbft/types/time"
+	cmn "github.com/tendermint/tendermint/libs/common" //nolint:depguard
 )
 
 // Canonical* wraps the structs in types for amino encoding them for use in SignBytes / the Signable interface.
 
 // TimeFormat is used for generating the sigs
 const TimeFormat = time.RFC3339Nano
 
+// TODO(@raneet10): Check whether we should keep it
+type CanonicalBlockID struct {
+	Hash        cmtbytes.HexBytes
+	PartsHeader CanonicalPartSetHeader
+}
+
+// TODO(@raneet10): Check whether we should keep it
+type CanonicalPartSetHeader struct {
+	Hash  cmtbytes.HexBytes
+	Total uint32
+}
+
+// TODO(@raneet10): Check whether removing Data field affects serialization
+type CanonicalProposal struct {
+	Type      SignedMsgType // type alias for byte
+	Height    int64         `binary:"fixed64"`
+	Round     int64         `binary:"fixed64"`
+	POLRound  int64         `binary:"fixed64"`
+	BlockID   CanonicalBlockID
+	Timestamp time.Time
+	ChainID   string
+	Data      cmn.HexBytes // [peppermint] add data in proposal to that signers can sign
+}
+
+// TODO(@raneet10): Check whether removing Data and SideTxResults fields affect serialization
+type CanonicalVote struct {
```

Comment:
And this one too (unused in current code)

Also,

> Check whether removing Data and SideTxResults fields affect serialization

This would be an issue if we went for DB migration. If we are doing hard-fork, then it's a non-issue

Reply:
Resolved